Source/MLX/Device.swift

```diff
@@ -94,3 +94,22 @@ extension Device: CustomStringConvertible {
         mlx_describe(ctx) ?? String(describing: type(of: self))
     }
 }
+
+/// Execute a block of code using a specific device.
+///
+/// Example:
+/// ```swift
+/// using(device: .gpu) {
+///    // code here will run on the GPU
+/// }
+/// ```
+///
+/// - Parameters:
+///     - device: device to be used
+///     - fn: function to be executed
+public func using<R>(device: Device, fn: () throws -> R) rethrows -> R {
```

Comment:
it's a free function, let me know if it's better to make it static function in `Device` class

Reply:
I think this looks good.